### PR TITLE
feat: link to service journey details from a trip suggestion

### PR DESCRIPTION
**Pull request by mortennordseth**
This PR adds the functionality to see details about the intermediate stops of a service journey from a trip pattern. 

I've added the headers referer to the global data so that we have the correct back link in the departures details page. If a user navigates to the departures details page from a trip suggestion one should be redirected back to the trip suggestion. In other cases (from departures or opening the link from someone) one should be navigated to the departures page. 

Closes https://github.com/AtB-AS/kundevendt/issues/15872

### CODE REVIEW

File: src/modules/cookies/index.tsx [hunk 1]
```diff
@@ -29,5 +30,6 @@ export function getGlobalCookies(req?: {
       language: req?.cookies[LANGUAGE_COOKIE_NAME] ?? null,
     },
     headersAcceptLanguage: req?.headers['accept-language'] ?? '',
+    referer: req?.headers.referer ?? '',
```
Comment (adriansberg) [inline src/modules/cookies/index.tsx:33]: Do we need to set it as a cookie?

Comment (mortennordseth) [reply src/modules/cookies/index.tsx:33]: I might be reading the code wrong here, but this shouldn't set it as a cookie. A bit weird naming of the function maybe. Should probably be "getInitialData" or something

File: src/page-modules/assistant/details/trip-section/estimated-calls-section.tsx [hunk 2]
```diff
@@ -2,30 +2,52 @@ import { Typo } from '@atb/components/typography';
...
+  serviceJourneyId,
+  date,
+  fromQuayId,
 }: EstimatedCallsSectionProps) {
   const { t, language } = useTranslation();
 
+  const shouldShowLinkToServiceJourney = serviceJourneyId ? true : false;
+  const serviceJourneyHref = fromQuayId
+    ? `/departures/details/${serviceJourneyId}?date=${date}&fromQuayId=${fromQuayId}`
+    : `/departures/details/${serviceJourneyId}?date=${date}`;
```
[the remark this answers is not in the document]
Comment (mortennordseth) [reply src/page-modules/assistant/details/trip-section/estimated-calls-section.tsx:29]: True, I'll fix 👍

File: src/modules/cookies/index.tsx (hunk 1 above)
Comment (adriansberg) [reply src/modules/cookies/index.tsx:33]: Nevermind me, they aren't set as a cookie, only read from cookie of course. Confused by the type and naming of function as well.

File: src/page-modules/departures/details/index.tsx [hunk 3]
```diff
@@ -49,13 +51,22 @@ export function DeparturesDetails({
     .map((s) => s.situationNumber)
     .filter((s): s is string => !!s);
 
+  const backLink =
+    referer && referer.includes('assistant/')
+      ? referer
+      : `/departures/${focusedCall.quay.stopPlace.id}`;
```
Comment (adriansberg) [inline src/page-modules/departures/details/index.tsx:57]:
```suggestion
  const backLink = referer?.includes('assistant/')
    ? referer
    : `/departures/${focusedCall.quay.stopPlace.id}`;
```

**Review by adriansberg — APPROVED** [review]

**Pull request merged**